### ISSUE TITLE
[BUG]WIN11 24H2 Chrome can not paly YouTube Video and more than half apps can not work

### ISSUE THREAD

**Issue by AnkiYuan** [MustardChef/WSABuilds#478]
### Describe the bug

WIN11 24H2 Chrome can not paly YouTube Video and more than half apps can not work

### Steps to reproduce the issue

just run apps

### Expected behaviour

fix

### Downloaded Build Of WSA

WSABuilds LTS Build #4 (15/07/24 - 2407.40000.0.0) for Windows 11 x64

### Windows build number

26100.2454

### PC Specification

设备名称	Anki-PC
处理器	AMD Ryzen 5 3600 6-Core Processor                 3.60 GHz
机带 RAM	32.0 GB
设备 ID	06B4B5CE-40FB-4686-8B25-0FFF3CDDB826
产品 ID	00391-70000-00000-AA783
系统类型	64 位操作系统, 基于 x64 的处理器
笔和触控	没有可用于此显示器的笔或触控输入

**Comment by MustardChef** [comment]
Update to the latest Non-LTS and LTS Builds.

**Comment by MustardChef** [comment]
> The official version 2407.40000.0.0 binary translator will stop working after 2025/6/1 (2311.40000.5.0 2024/12/1)

Is this true? Where did you get this information from? Source?

**Comment by AnkiYuan** [comment]
Tried the newest Windows Subsystem For Android 2407.40000.0.0 for Windows 11 x64 ,It's works. Thanks

**Comment by MustardChef** [comment]
> > Where did you get this information from? Source?
> 
> Completely shutdown WSA, set computer time to June 2, 2025

You are right. The ARM translation layer stops working when the date is set.

**Comment by AnkiYuan** [comment]
So still need to wait the last ver?

**Comment by chip33** [comment]
> > > Where did you get this information from? Source?
> > 
> > 
> > Completely shutdown WSA, set computer time to June 2, 2025
> 
> You are right. The ARM translation layer stops working when the date is set. I guess that they did indeed place a time bomb in software.

This is pretty big news, I don't see it reported anywhere else. I guess there is no good way round this?

**Comment by FocusBlast** [comment]
Hey! Could there be an old build without this timebomb and we could still use it until Google pulls support for that specific Android version? I really don't want WSA to go away ;___;

**Comment by hifihedgehog** [comment]
> > Where did you get this information from? Source?
> 
> Completely shutdown WSA, set computer time to June 2, 2025

As a counterpoint, I just did this. Rebooted my system several times to confirm. WSA still works. @MustardChef, you may want to verify this. Maybe this was legitimate at one point. Who knows? Maybe Microsoft removed the time bomb in a rare act of mercy? That, or the report was erroneous.

![Image](https://github.com/user-attachments/assets/0a00d465-6826-4161-9523-e2cdda208647)

**Comment by robbely** [comment]
> > > Where did you get this information from? Source?
> > 
> > 
> > Completely shutdown WSA, set computer time to June 2, 2025
> 
> As a counterpoint, I just did this. Rebooted my system several times to confirm. WSA still works. [@MustardChef](https://github.com/MustardChef), you may want to verify this. Maybe this was legitimate at one point. Who knows? Maybe Microsoft removed the time bomb in a rare act of mercy? That, or the report was erroneous.
> 

After the timebomb, WSA will still start up and run a lot of apps, there is a large number of apps that will stop running though. Try running youtube in chrome, or starting up the Shazam app

**Comment by hifihedgehog** [comment]
> youtube in chrome

Now this makes sense. First though. Outside of this as an academic example, why would I even want to run YouTube in Chrome? I would expend unnecessary CPU cycles compared to running Edge natively in Windows.

I am trying to reproduce your issue and cannot. Here is why. You and others are misidentifying an issue. Even Windows programs, even a simple web browser, will stop working correctly without a proper local time on the system clock. Valid system clock time is required for Internet activity and the two apps you mentioned need web connectivity to work. If I set the time on an Android tablet ahead by several months, it also fails to connect to the Internet. Here is an example below of what happens in any browser in Windows if I have the machine's local time incorrect.

![Image](https://github.com/user-attachments/assets/316e92fb-0044-426d-a52d-8d1f2caa454a)

**Comment by hifihedgehog** [comment]
> The official version 2407.40000.0.0 binary translator will stop working after 2025/6/1 (2311.40000.5.0 2024/12/1)

This is false.

**Comment by robbely** [comment]
> > youtube in chrome
> 
> Now this makes sense.
> 
> First though. Outside of this as an academic example, why would I even want to run YouTube in Chrome in WSA? I would expend unnecessary CPU cycles compared to running Edge natively in Windows.
> 
> I am trying to reproduce your issue and cannot. Here is why. You and others are misidentifying an issue. Even Windows programs, even a simple web browser, will stop working correctly without a proper local time on the system clock. Valid system clock time is required for Internet activity (HTTPS needs it, for example, for establishing the validity of an SSL/TLS certificate) and the two apps you mentioned need web connectivity to work. If I set the time on an Android tablet ahead by several months, it also fails to connect to the Internet. Here is an example below of what happens in any browser in Windows if I have the machine's local time incorrect.


Youtube in Chrome was the OP's experience (I quoted that as they didn't give any other specific apps). From some experiments some users did in the discord server, changing the date before June didn't break anything, changing it after did. I haven't tried this myself but enough users did to convince me. Hopefully they're wrong and you're right ;) we shall have to wait and see.

**Comment by MustardChef** [comment]
> > youtube in chrome
> 
> Now this makes sense.
> 
> First though. Outside of this as an academic example, why would I even want to run YouTube in Chrome in WSA? I would expend unnecessary CPU cycles compared to running Edge natively in Windows.
> 
> I am trying to reproduce your issue and cannot. Here is why. You and others are misidentifying an issue. Even Windows programs, even a simple web browser, will stop working correctly without a proper local time on the system clock. Valid system clock time is required for Internet activity (HTTPS needs it, for example, for establishing the validity of an SSL/TLS certificate) and the two apps you mentioned need web connectivity to work. If I set the time on an Android tablet ahead by several months, it also fails to connect to the Internet. Here is an example below of what happens in any browser in Windows if I have the machine's local time incorrect.
> 
> ![Image](https://github.com/user-attachments/assets/316e92fb-0044-426d-a52d-8d1f2caa454a)

Could you try an ARM app and see if it is working?

**Comment by raaomoka** [comment]
Hah, guess this time bomb is why all of a sudden none of my Android apps were opening on WSA v2310.40000.2.0 🤦 Here's hoping we find some means to remove the stupid time bomb artificially dictating what can run on our systems...

**Comment by zaphod77** [comment]
This time bomb was probably put in by intel, to make sure that MS continued to pay up to license libhoudini for their paid operating system.  MS doesn't want to pay Intel anymore, so it's going to stop working.

Solutions include
1) switch to libndk, but that only works well for AMD cpus, it's horrid on intel by all accounts.
2) find somewhere else to yoink a working no timebomb libhoudini from, like maybe Google Play Games

**Comment by robbely** [comment]
According to the discord server, mustardchef has a plan for the upcoming timebomb - lets hope its a good fix

**Comment by raaomoka** [comment]
Not sure if this will be of any help, but even after updating to version 2407.40000.4.0, attempting to start up applications still didn't work. Thought it was requiring me to restart, but I couldn't do that at the moment (in the middle of some other work).

Out of curiosity, I just switched the WSA **Advanced settings > Experimental features > Vulkan driver** from **System Default** to **D3D12**, then reattempted opening the Play Store. Surprisingly, everything is up and running again even without a system restart. Here's hoping it stays that way 🤞

**Comment by chip33** [comment]
> Out of curiosity, I just switched the WSA **Advanced settings > Experimental features > Vulkan driver** from **System Default** to **D3D12**, then reattempted opening the Play Store. Surprisingly, everything is up and running again

I can confirm that switching to D3D12 fixes everything for me too on latest win 11 insider beta 26120.3872
